### PR TITLE
[NIFI-10876] Made changes to convert from JUnit 4 to JUnit 5

### PR DESCRIPTION
**Pull request by dan-s1**
# Summary

[NIFI-10876](https://issues.apache.org/jira/browse/NIFI-10876)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java [hunk 1]
```diff
@@ -119,7 +123,7 @@ public void testAddPoliciesWithSameResourceAndAction() {
 
         try {
             accessPolicyProvider.addAccessPolicy(policy2);
-            Assert.fail("Should have thrown exception");
+            fail("Should have thrown exception");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java:126]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java [hunk 2]
```diff
@@ -142,7 +146,7 @@ public void testAddUsersWithSameIdentity() {
 
         try {
             userGroupProvider.addUser(user2);
-            Assert.fail("Should have thrown exception");
+            fail("Should have thrown exception");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java:149]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java [hunk 3]
```diff
@@ -165,7 +169,7 @@ public void testAddGroupsWithSameName() {
 
         try {
             userGroupProvider.addGroup(group2);
-            Assert.fail("Should have thrown exception");
+            fail("Should have thrown exception");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java:172]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java [hunk 4]
```diff
@@ -224,7 +228,7 @@ public void testUpdateUserWithSameIdentity() {
         try {
             User user1Updated = new User.Builder().identifier("user-id-1").identity("xyz").build();
             userGroupProvider.updateUser(user1Updated);
-            Assert.fail("Should have thrown exception");
+            fail("Should have thrown exception");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java:231]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java [hunk 5]
```diff
@@ -249,7 +253,7 @@ public void testUpdateGroupWithSameName() {
         try {
             Group group1Updated = new Group.Builder().identifier("group-id-1").name("xyz").build();
             userGroupProvider.updateGroup(group1Updated);
-            Assert.fail("Should have thrown exception");
+            fail("Should have thrown exception");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-authorizer/src/test/java/org/apache/nifi/authorization/AuthorizerFactoryTest.java:256]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/controller/repository/TestRingBufferEventRepository.java [hunk 6]
```diff
@@ -57,21 +57,21 @@ public void testPurge() throws IOException {
         repo.updateRepository(generateEvent(), id2);
         RepositoryStatusReport report = repo.reportTransferEvents(System.currentTimeMillis());
         FlowFileEvent entry = report.getReportEntry(id1);
-        Assert.assertNotNull(entry);
+        Assertions.assertNotNull(entry);
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/controller/repository/TestRingBufferEventRepository.java:60]: Let's move to static imports here for consistency.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizerTest.java [hunk 7]
```diff
@@ -552,7 +552,7 @@ public void testWaitForQueueToEmpty() throws InterruptedException {
                 synchronizer.synchronize(connectionAB, null, group, synchronizationOptions);
                 completionLatch.countDown();
             } catch (final Exception e) {
-                Assert.fail(e.toString());
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizerTest.java:555]: Should used `assertDoesNotThrow`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java [hunk 8]
```diff
@@ -231,7 +232,7 @@ public void testChangingSensitivity() {
 
         try {
             context.setParameters(updatedParameters);
-            Assert.fail("Succeeded in changing parameter from non-sensitive to sensitive");
+            fail("Succeeded in changing parameter from non-sensitive to sensitive");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java:235]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java [hunk 9]
```diff
@@ -241,7 +242,7 @@ public void testChangingSensitivity() {
 
         try {
             context.setParameters(updatedParameters);
-            Assert.fail("Succeeded in changing parameter from sensitive to non-sensitive");
+            fail("Succeeded in changing parameter from sensitive to non-sensitive");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java:245]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java [hunk 10]
```diff
@@ -281,7 +282,7 @@ public void testChangingParameterForRunningProcessor() {
         parameters.put("abc", new Parameter(abcDescriptor, null));
         try {
             context.setParameters(parameters);
-            Assert.fail("Was able to remove parameter while referencing processor was running");
+            fail("Was able to remove parameter while referencing processor was running");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java:285]: Should use `assertThrows`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java [hunk 11]
```diff
@@ -331,17 +332,17 @@ public void testChangingNestedParameterForRunningProcessor() {
         startProcessor(procNode);
         try {
             a.setInheritedParameterContexts(Collections.emptyList());
-            Assert.fail("Was able to remove parameter while referencing processor was running");
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java:334]: This should be replaced with something like `IllegalStateException ex = assertThrows(...`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/util/TestFlowDifferenceFilters.java [hunk 12]
```diff
@@ -62,7 +62,7 @@ public void testFilterAddedRemotePortsWithRemoteOutputPort() {
                 DifferenceType.COMPONENT_ADDED, null, remoteGroupPort, null, null, "");
 
         // predicate should return false because we don't want to include changes for adding a remote input port
-        Assert.assertFalse(FlowDifferenceFilters.FILTER_ADDED_REMOVED_REMOTE_PORTS.test(flowDifference));
+        Assertions.assertFalse(FlowDifferenceFilters.FILTER_ADDED_REMOVED_REMOTE_PORTS.test(flowDifference));
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/util/TestFlowDifferenceFilters.java:65]: Static imports

**Review by MikeThomsen — CHANGES_REQUESTED** [review]
`fail` calls should be replaced with appropriate use of `assertThrows/assertDoesNotThrow`

**Comment by MikeThomsen** [comment]
@dan-s1 You also have a merge conflict. Do a rebase (ie don't do merge main into your branch).

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/test/java/org/apache/nifi/util/CharacterFilterUtilsTest.java [hunk 13]
```diff
@@ -27,7 +27,7 @@ public void filterInvalidCharacters() throws Exception {
         final String filtered = CharacterFilterUtils.filterInvalidXmlCharacters(text);
 
         final String expected = "This is an example with characters that need to be filtered  in it. ";
-        Assert.assertEquals(expected, filtered);
+        Assertions.assertEquals(expected, filtered);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/test/java/org/apache/nifi/util/CharacterFilterUtilsTest.java:30]: This is a good opportunity to replace class references with static imports:
```suggestion
        assertEquals(expected, filtered);
```

**Comment by exceptionfactory** [comment]
@dan-s1 After rebasing you will need to use `--force` with `git push` to override the remote version with your current local version.

**Comment by dan-s1** [comment]
> @dan-s1 After rebasing you will need to use `--force` with `git push` to override the remote version with your current local version.

Thanks! Done.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/controller/repository/TestRingBufferEventRepository.java [hunk 14]
```diff
@@ -17,14 +17,17 @@
 package org.apache.nifi.controller.repository;
 
 import org.apache.nifi.controller.repository.metrics.RingBufferEventRepository;
-import org.junit.Assert;
 import org.junit.Test;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/controller/repository/TestRingBufferEventRepository.java:20]: This needs to be changed to JUnit Jupiter.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java [hunk 15]
```diff
@@ -35,10 +34,12 @@
 import java.util.Set;
 
 import static junit.framework.TestCase.assertTrue;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java:36]: This should be replaced with the Jupiter Assertions import.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java [hunk 16]
```diff
@@ -20,7 +20,6 @@
 import org.apache.nifi.controller.service.ControllerServiceNode;
 import org.apache.nifi.controller.service.ControllerServiceState;
 import org.apache.nifi.groups.ProcessGroup;
-import org.junit.Assert;
 import org.junit.Test;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/parameter/TestStandardParameterContext.java:23]: This should be replaced with JUnit Jupiter Test

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the updates @dan-s1, this looks just about ready to go. I noted three places where old imports need to be replaced with JUnit Jupiter.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the updates @dan-s1, it looks like the latest commit build failed on the following unit test:

```
Error:  org.apache.nifi.authorization.AuthorizerFactoryTest.testUpdateUserWithSameIdentity  Time elapsed: 0.511 s  <<< ERROR!
java.lang.IllegalStateException: User/user group already exists with the identity 'xyz'.
	at org.apache.nifi.authorization.AuthorizerFactory$1$1$1.updateUser(AuthorizerFactory.java:253)
	at org.apache.nifi.authorization.AuthorizerFactoryTest.testUpdateUserWithSameIdentity(AuthorizerFactoryTest.java:216)
```

**Comment by dan-s1** [comment]
> Thanks for the updates @dan-s1, it looks like the latest commit build failed on the following unit test:
> 
> ```
> Error:  org.apache.nifi.authorization.AuthorizerFactoryTest.testUpdateUserWithSameIdentity  Time elapsed: 0.511 s  <<< ERROR!
> java.lang.IllegalStateException: User/user group already exists with the identity 'xyz'.
> 	at org.apache.nifi.authorization.AuthorizerFactory$1$1$1.updateUser(AuthorizerFactory.java:253)
> 	at org.apache.nifi.authorization.AuthorizerFactoryTest.testUpdateUserWithSameIdentity(AuthorizerFactoryTest.java:216)
> ```

Fixed. I forgot to remove those lines after I placed in assertThrows.

**Comment by dan-s1** [comment]
@exceptionfactory Look like all tests have passed now

**Review by exceptionfactory — APPROVED** [review]
Thanks for working through the feedback and making these improvements @dan-s1, the latest version looks good! +1 merging

**Pull request closed without merging**